gpbft/chain.go

```diff
@@ -45,8 +105,8 @@ func (c ECChain) IsZero() bool {
 }
 
 // Returns the base tipset.
-func (c ECChain) Base() TipSet {
-	return c[0]
+func (c ECChain) Base() *TipSet {
```

Comment:
I'm returning by pointer here and elsewhere as this structure isn't small.

Reply:
I'm also happy to bikeshed on this (e.g., take by pointer in more places to reduce the dereferencing, etc.).

Reply:
Yeah this interface was mostly pointers before I stripped out the type, if I recall. Should probably have more pointer receivers and parameters in that case.